pwiz_tools/Skyline/Model/AuditLog/AuditLogEntry.cs

```diff
@@ -1104,20 +1181,150 @@ public void ReadXml(XmlReader reader)
             Reason = reader.IsStartElement(EL.reason) ? reader.ReadElementString() : string.Empty;
             ExtraInfo = reader.IsStartElement(EL.extra_info) ? reader.ReadElementString().UnescapeNonPrintableChars() : string.Empty;
 
-            UndoRedo = reader.DeserializeElement<LogMessage>().ChangeLevel(LogLevel.undo_redo);
-            Summary = reader.DeserializeElement<LogMessage>().ChangeLevel(LogLevel.summary);
+            EL allInfoEnum;
+            if (reader.IsStartElement(EL.undo_redo))
+            {
+                UndoRedo = reader.DeserializeElement<LogMessage>(EL.undo_redo).ChangeLevel(LogLevel.undo_redo);
+                Summary = reader.DeserializeElement<LogMessage>(EL.summary).ChangeLevel(LogLevel.summary);
+                allInfoEnum = EL.all_info;
+            }
+            else // Backward compatibility
+            {
+                UndoRedo = reader.DeserializeElement<LogMessage>(EL.message).ChangeLevel(LogLevel.undo_redo);
+                Summary = reader.DeserializeElement<LogMessage>(EL.message).ChangeLevel(LogLevel.summary);
+                allInfoEnum = EL.message;
+            }
 
             var list = new List<LogMessage>();
-            while (reader.IsStartElement(EL.message))
-                list.Add(reader.DeserializeElement<LogMessage>().ChangeLevel(LogLevel.all_info));
+            while (reader.IsStartElement(allInfoEnum))
+                list.Add(reader.DeserializeElement<LogMessage>(allInfoEnum).ChangeLevel(LogLevel.all_info));
 
             AllInfo = list;
 
+            // Think about how and if we want to store these english strings
+            EnExtraInfo = reader.IsStartElement(EL.en_extra_info)
```

Comment:
Normally with something like this, I would not use the stored value since it can be recalculated, but I guess since you are checking the hash to make sure it is exactly the same as what was written, it may not matter. What happens if the hash doesn't match? It seems like you could test whether the hash would match with the generated text to see if someone only tampered with the english text.

Reply:
If the hashes are different I am already recalculating the hashes of the log messages and the extra info, I haven't push that code yet. (I actually just set them to null so that they only get calculated when we actually use them for the first time)